### PR TITLE
Add spider for Mossoró/RN

### PR DESCRIPTION
**Pull request by agessner**
In this PR we add the spider to collect data from [Mossoró/RN gazettes](https://www.prefeiturademossoro.com.br/jom/alljoms.php).

We have found that the gazette's dates in the website are not  trustworthy:
- We have a gazette with a date that doesn't exist, `June 31st, 2017`. But inside the gazette the date is `May 31st, 2017`. Right now, it breaks the code when it tries to parse the date.
How should we proceed? We thought about reading the gazette content and get the date from there. But right now we couldn't find a way to do that. Any thoughts?

- We have dates in future, like `December 17th, 2018`.  But inside the gazette the date is `January 17th, 2018`. It is similar to the previous problem.

- We have gazettes that are duplicated, like 2018-01-12, 2017-12-20, 2017-10-31, etc...
About that, we thought that we could check if a given gazette's date has the same date from another gazette that was already processed, if so we ignore it. Is it a good approach?

Co-authored-by: @LucasSedrez and @rhogeranacleto

### CODE REVIEW

**Comment by giovanisleite** [comment]
Hey, nice contribution @agessner, Lucas and Rogher :smiley: 

> We have a gazette with a date that doesn't exist, June 31st, 2017. But inside the gazette the date is May 31st, 2017. Right now, it breaks the code when it tries to parse the date.
How should we proceed? We thought about reading the gazette content and get the date from there. But right now we couldn't find a way to do that. Any thoughts?

To do that you have to change the pipeline and create a custom one to extract the date from pdf text, I don't know how to do it, but @rennerocha can give instructions here, i guess

> We have dates in future, like December 17th, 2018. But inside the gazette the date is January 17th, 2018. It is similar to the previous problem.

The first point solution solves this one too.

> We have gazettes that are duplicated, like 2018-01-12, 2017-12-20, 2017-10-31, etc...
About that, we thought that we could check if a given gazette's date has the same date from another gazette that was already processed, if so we ignore it. Is it a good approach?

I think this will get a database error and let it go, so, don't bother.

File: processing/data_collection/gazette/spiders/rn_mossoro.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            url = self.extract_url(element)
+            date = self.extract_date(element)
+            if url and date and date not in dates:
+                dates.add(date)
+                yield Gazette(
+                    date=date,
+                    file_urls=[url],
+                    is_extra_edition=self.is_extra_edition(element),
+                    municipality_id=self.MUNICIPALITY_ID,
+                    power='executive_legislature',
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/rn_mossoro.py]: Looking some gazettes, I saw just executive information. So, I coudn't confirm that.
Did you saw something from the city council (legislature)?

File: processing/data_collection/gazette/spiders/rn_mossoro.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    scraped_at=dt.datetime.utcnow(),
+                )
+
+    def extract_url(self, element):
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/rn_mossoro.py]: Selectors working =)

File: processing/data_collection/gazette/spiders/rn_mossoro.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class RnMossoroSpider(BaseGazetteSpider):
+    MUNICIPALITY_ID = '2408003'
+    PDF_URL = 'https://www.prefeiturademossoro.com.br/jom/{}'
+
+    GAZETTES_CSS = 'body > div > table:first-child > tr > td > table:last-child > tr > td > div > table > tr > td'
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/rn_mossoro.py]: (optional) remove ```body > div > ```

File: processing/data_collection/gazette/spiders/rn_mossoro.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    PDF_INFO_CSS = 'td.TextoLink'
+    PDF_REALTIVE_PATH_CSS = 'a::attr(href)'
```
Comment (mvinoba) [inline processing/data_collection/gazette/spiders/rn_mossoro.py]: Did you mean to spell `relative`? This same spelling repeats along the file.

File: processing/data_collection/gazette/spiders/rn_mossoro.py (hunk 1 above)
Comment (agessner) [reply processing/data_collection/gazette/spiders/rn_mossoro.py]: Well, we saw some gazettes like this one https://www.prefeiturademossoro.com.br/jom/jom264.pdf with a legislative section. That's why we set it as `executive_legislature`.

File: processing/data_collection/gazette/spiders/rn_mossoro.py (hunk 4 above)
Comment (agessner) [reply processing/data_collection/gazette/spiders/rn_mossoro.py]: I did. It was a misspelling, sorry about that.

File: processing/data_collection/gazette/spiders/rn_mossoro.py (hunk 3 above)
Comment (agessner) [reply processing/data_collection/gazette/spiders/rn_mossoro.py]: done :+1:

**Comment by vitorbaptista** [comment]
@agessner I updated your code with the latest in `main`, and fixed the issues caused by a change on Mossoró's website. The new PR is https://github.com/okfn-brasil/querido-diario/pull/296. I'm closing this now, as all the code from here is in that new PR.

**Pull request closed without merging**